config/daemonconfig.go

```diff
@@ -168,7 +168,11 @@ func NewDaemonConfig(fsDriver string, cfg DaemonConfig, imageID, snapshotID stri
 			backendConfig = &cfg.Config.BackendConfig
 			fscacheID := erofs.FscacheID(snapshotID)
 			cfg.ID = fscacheID
-			cfg.DomainID = fscacheID
+			if len(SharedDomainID) == 0 {
+				cfg.DomainID = fscacheID
+			} else {
+				cfg.DomainID = SharedDomainID
+			}
```

Comment:
The logic of getting and parsing `shared_domain_id` is a little dim and tedious. IIUC, the user is wanting the shared domain feature by adding an item to the nydusd configuration like `shared_domain_id=blabla`, so it can be loaded from the json file rather than read from a global variable at compiling time. We don't even need to touch the item but just pass it to the in-memory fscache configuration and use it when perform EROFS mount.  So the assignment `cfg.DomainID = SharedDomainID` here is not necessary